### ISSUE TITLE
Implement a variadic interface instead of fields

### ISSUE THREAD

**Issue by sagikazarmark** [goph/logur#25]
Currently log functions accept a `map[string]interface{}` as the second argument which is far from ideal:

1. It's cumbersome to write
2. It's a mandatory argument
3. Merging context is slow this way (compared to slices)

Consider switching to a variadic slice of key-value pairs.

Some pro-contra arguments are listed here: https://github.com/goph/logur#why-not-loggerwithkeyvals-interface

**Comment by sagikazarmark** [comment]
Using key-value pairs would become much easier with generics

**Comment by ghostsquad** [comment]
@sagikazarmark have you looked at using https://github.com/cheekybits/genny ?

I am very interested in this, and I know it's one opinion vs another, but in your readme, you talk about how "order matters" to some people. For me, in development, anything to make reading the logs easier is a benefit. In production, order doesn't matter, as that logs are json encoded and shipped off elsewhere.

**Comment by sagikazarmark** [comment]
No, I haven't, but I'm afraid it would make the interface less portable.

Order in this case means the order fields are added. Nothing keeps you from sorting them by name for example on output. In fact, I think Logrus does exactly that.

**Comment by ghostsquad** [comment]
OH! yes, sorting by name is a quick and dirty (but acceptable) hack. As long as the same field appears in the same location reliably, I don't really care otherwise. :)
